### ISSUE TITLE
weights become very large and then loss = nan

### ISSUE THREAD

**Issue by chyh1990** [BVLC/caffe#373]
I use caffe to train my CNN but loss become nan after a few thousand iterations.
I dump the weights before that iterations and I found that some weights in the inner product layers become very large (e.g. +3e29). I check my shuffled input data and make sure they are in reasonable range. 

I represent this in both GPU mode and CPU mode. Does this indicate some numeric problems in caffe or other causes?

**Comment by henurgwpf** [comment]
@chyh1990 Could you tell me how to get all parameters in my training model? Thanks very much!

**Comment by hyojinie** [comment]
I understand that regularization gets rid of unreasonable values for biases and weights, but I wonder how could a loss become NaN? Is it because those values resulted the loss to be -inf?
